dtm0/fop.c

```diff
@@ -221,10 +171,22 @@ static int dtm0_fom_create(struct m0_fop *fop,
 	/* see ::m0_dtm0_dtx_post_persistent for the details */
 	fop->f_opaque = pma;
 	reply = m0_fop_data(repfop);
+	req   = m0_fop_data(fop);
 	reply->dr_txr = (struct m0_dtm0_tx_desc) {};
 	reply->dr_rc = 0;
-	m0_fom_init(&fom->dtf_fom, &fop->f_type->ft_fom_type,
-		    &dtm0_req_fom_ops, fop, repfop, reqh);
+
+	if (req->dtr_msg == DTM_EXECUTE) {
+		M0_ASSERT_INFO(m0_dtm0_in_ut(), "Emsg FOM is only for UTs.");
```

Comment:
Yes. I added a comment to explain it a bit in https://github.com/Seagate/cortx-motr/pull/802/commits/c75464d7d2ce0553ee63b8fa00745944e6dee137#diff-7d29e2d404e16606678795ff91d5da8bc5dbc20695239029de046a0ac8bb622dR363

Essentially, this is dead code. It is not in use anywhere. However, once we get drlink ready to be used in m0_rpc_post_sync-way, we will enable the UT back (dtm0-service UT) so that this tick will be in use. I did not remove this code (as well as the corresponding UT code) because I'm sure we will resurrect it (well, we _have to_ resurrect it -- we need UTs for drlink).